### PR TITLE
Add code to allow offline processing of 2D Cartographer bags.

### PR DESCRIPTION
**Pull request by wohe**
Fixes waiting for transforms to be published, adds support for multi-echo
laser scans, sets default values for the submap topic and query for the
RViz plugin.

### CODE REVIEW

File: cartographer_ros/launch/mapping_2d.launch [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<!--
```
Comment (damonkohler) [inline cartographer_ros/launch/mapping_2d.launch]: Please rename to backpack_2d.launch in a followup CL.

Comment (wohe) [inline cartographer_ros/launch/mapping_2d.launch]: Ack.

File: cartographer_ros/src/submaps_display.cc [hunk 2]
```diff
@@ -136,7 +138,7 @@ void SubmapsDisplay::onInitialize() {
       kSubmapTexturesGroup);
 
   scene_manager_->addListener(&scene_manager_listener_);
-  UpdateTopic();
+  UpdateSubmapQueryServiceName();
```
Comment (damonkohler) [inline cartographer_ros/src/submaps_display.cc]: Please add a todo to combine this with UpdateTopic(). It's unlikely we ever want to do one without doing the other as well.

File: cartographer_ros/urdf/backpack_2d.urdf [hunk 3]
```diff
@@ -66,20 +66,20 @@
   </link>
 
   <joint name="base_link_joint" type="fixed">
```
Comment (damonkohler) [inline cartographer_ros/urdf/backpack_2d.urdf]: imu_link_joint

File: cartographer_ros/src/submaps_display.cc (hunk 2 above)
Comment (wohe) [inline cartographer_ros/src/submaps_display.cc]: Done.

File: cartographer_ros/urdf/backpack_2d.urdf (hunk 3 above)
Comment (wohe) [inline cartographer_ros/urdf/backpack_2d.urdf]: Done.

**Comment by damonkohler** [comment]
LGTM

**Pull request merged**